### PR TITLE
Check time

### PR DESCRIPTION
**Pull request by AGradowski**

### CODE REVIEW

File: contracts/contracts/Rental.sol [hunk 1]
```diff
@@ -21,6 +22,12 @@ contract Rental {
         RentalStatus rentalState;// is this start or end of rental
     }
 
+    //used to get start reord and end record at the same time
```
Comment (JanuszJakubiec) [inline contracts/contracts/Rental.sol:25]:
```suggestion
    //used to get start record and end record at the same time
```

File: contracts/contracts/Tarrif.sol [hunk 2]
```diff
@@ -1,15 +1,26 @@
...
+  enum TarrifStatus{ACTIVE, ENDED}
+  
+
+  struct TarrifEntry {
         uint256 carTypeID;//from the carDB
         uint256 pricePerMinute;//or per kilometer, to discuss, or add more
         uint256 blockchainTime;//when this entry was added (from when its valid)
-    }
+        TarrifStatus status;
+
```
Comment (JanuszJakubiec) [inline contracts/contracts/Tarrif.sol:13]:
```suggestion
```

File: contracts/contracts/Tarrif.sol [hunk 3]
```diff
@@ -1,15 +1,26 @@
...
 pragma solidity >=0.4.22 <0.9.0;
 
-contract Tarrif{
-    address private ADMIN_ADDRESS = msg.sender;//the address that put this contract here (though truffle it is the index 0 address)
-    
-    struct TarrifEntry {
+interface Tarrif {
+  enum TarrifStatus{ACTIVE, ENDED}
+  
+
```
Comment (JanuszJakubiec) [inline contracts/contracts/Tarrif.sol:7]:
```suggestion
```

File: contracts/contracts/Tarrif.sol [hunk 4]
```diff
@@ -18,29 +29,50 @@ contract Tarrif{
...
+      for(uint256 i = tarrifPricing.length; i>0; i--){
+        if(tarrifPricing[i-1].carTypeID == _carTypeID && tarrifPricing[i-1].status == TarrifStatus.ACTIVE){
+          tarrifPricing[i-1].status = TarrifStatus.ENDED;
+        }
+      }
+      revert("No entry for that carTypeID");
+    }
 
-    function getPricing() public view returns(uint256[] memory,uint256[] memory,uint256[] memory){
+    function getFullPricing() external view returns(uint256[] memory,uint256[] memory,uint256[] memory){
```
Comment (JanuszJakubiec) [inline contracts/contracts/Tarrif.sol:51]:
```suggestion
    function getFullPricing() external view returns(uint256[] memory, uint256[] memory, uint256[] memory){
```

File: contracts/contracts/Tarrif.sol [hunk 5]
```diff
@@ -1,15 +1,26 @@
...
+  struct TarrifEntry {
         uint256 carTypeID;//from the carDB
         uint256 pricePerMinute;//or per kilometer, to discuss, or add more
         uint256 blockchainTime;//when this entry was added (from when its valid)
-    }
+        TarrifStatus status;
+
+  }
+
+  function getFullPricing() external view returns(uint256[] memory,uint256[] memory,uint256[] memory);
```
Comment (JanuszJakubiec) [inline contracts/contracts/Tarrif.sol:16]:
```suggestion
  function getFullPricing() external view returns(uint256[] memory, uint256[] memory, uint256[] memory);
```

File: contracts/test/test_rental.js [hunk 6]
```diff
@@ -1,11 +1,14 @@
...
+const TarrifContract = artifacts.require("TarrifContract");
 const chai = require('chai')
 const chaiAsPromised = require('chai-as-promised');
 
 contract("Rental", accounts => {
     const creatorAddress = accounts[0];
     before(async () => {
       this.rentalTest = await Rental.deployed();
+      this.tarrifHelp = await TarrifContract.deployed();
+      this.tarrifHelp.addEntry(1,10);
```
Comment (JanuszJakubiec) [inline contracts/test/test_rental.js:11]:
```suggestion
      this.tarrifHelp.addEntry(1, 10);
```

File: contracts/test/test_rental.js [hunk 7]
```diff
@@ -35,8 +38,8 @@ contract("Rental", accounts => {
     });
 
     it("should add active rental and get it", async () => {
-        const emit_event = await this.rentalTest.startRental(10,1,1);
-        
+       // await this.tarrifHelp.addEntry(1,10);
```
Comment (JanuszJakubiec) [inline contracts/test/test_rental.js:41]:
```suggestion
       // await this.tarrifHelp.addEntry(1, 10);
```

File: contracts/test/test_rental.js [hunk 8]
```diff
@@ -35,8 +38,8 @@ contract("Rental", accounts => {
...
+        const emit_event = await this.rentalTest.startRental(10,1,1,this.tarrifHelp.address,1);
```
Comment (JanuszJakubiec) [inline contracts/test/test_rental.js:42]:
```suggestion
        const emit_event = await this.rentalTest.startRental(10, 1, 1, this.tarrifHelp.address, 1);
```

File: contracts/test/test_rental.js [hunk 9]
```diff
@@ -67,7 +70,7 @@ contract("Rental", accounts => {
 
     it("should return two cars (array of three, with two marked as free (so false)), after one has been rented", async () => {
       await this.rentalTest.addCar(3);
-      await this.rentalTest.startRental(10,1,1);
+      await this.rentalTest.startRental(10,1,1,this.tarrifHelp.address,1);
```
Comment (JanuszJakubiec) [inline contracts/test/test_rental.js:73]:
```suggestion
      await this.rentalTest.startRental(10, 1, 1, this.tarrifHelp.address, 1);
```

File: rental/src/main/java/pl/edu/agh/rental/rental/RentalController.java [hunk 10]
```diff
@@ -41,7 +42,34 @@ public void endRental(@PathVariable("id") final Integer reservationId,
     @GetMapping()
     @ResponseStatus(HttpStatus.OK)
     public RentalData getRental(@CurrentUser final User user) throws NoRentalError {
-        //throw new NoRentalError();
         return reservationService.getRental(user);
     }
+
+    //cuurent time, current rental
```
Comment (JanuszJakubiec) [inline rental/src/main/java/pl/edu/agh/rental/rental/RentalController.java:48]:
```suggestion
    //current time, current rental
```

File: rental/src/main/java/pl/edu/agh/rental/rental/dto/RentalCreateInput.java [hunk 11]
```diff
@@ -4,5 +4,7 @@
 
 public record RentalCreateInput(
         @NotNull
-        Integer carId
-) { }
+        Integer carId,
+        Integer carTypeId
+) {
```
Comment (JanuszJakubiec) [inline rental/src/main/java/pl/edu/agh/rental/rental/dto/RentalCreateInput.java:9]:
```suggestion
) { }
```

File: rental/src/main/java/pl/edu/agh/rental/rental/dto/RentalCreateInput.java [hunk 12]
```diff
@@ -4,5 +4,7 @@
...
+}
```
Comment (JanuszJakubiec) [inline rental/src/main/java/pl/edu/agh/rental/rental/dto/RentalCreateInput.java:10]:
```suggestion
```

**Review by JanuszJakubiec — CHANGES_REQUESTED** [review]
Looks good, minor comments

File: contracts/test/test_rental.js (hunk 7 above)
Comment (AGradowski) [reply contracts/test/test_rental.js:41]: There should not be any lines commented like that

Comment (AGradowski) [reply contracts/test/test_rental.js]: Removed it, it is also called in the befre(async()) part

**Review by JanuszJakubiec — APPROVED** [review]

**Pull request merged**